### ISSUE TITLE
Stock up db for main characters

### ISSUE THREAD

**Issue by marcusnovotny** [Rostlab/JS16_ProjectD_Group4#119]
Seems like the first full crawl of csv.got.show is completed and key characters are still missing. We have about 1,5 weeks of data for:

http://csv.got.show/Daenerys_Targaryen
http://csv.got.show/Tywin_Lannister
http://csv.got.show/Tyrion_Lannister
http://csv.got.show/Jon_Snow

I don't know where this is coming from suddenly, but I'm having the same issue on my machine at home. Those characters used to top my Most Discussed list during the first crawl and now we barely get data on them.

Is there anything we can do to squeeze the remaining tweets out of Twitter? Is it possible they stop us from accessing them because we crawled a lot of data from these terms? It feels super strange to me.

**Comment by julienschmidt** [comment]
Retry, retry, retry. If you set the `found` to something like `10000` and the retry threshold to e.g. 500, the crawler will retry really aggressively. Sometimes it suddenly works then. I think Twitter has some really bad caching mechanism.

**Comment by julienschmidt** [comment]
And by retry I mean `gotsent.updateCharacter("<name>", true);`

**Comment by marcusnovotny** [comment]
Will check at home :+1:

**Comment by julienschmidt** [comment]
Or we might try to use all our old DB dumps to add missing tweets.

**Comment by marcusnovotny** [comment]
I think I only had that data in the very first one with the different character IDs

**Comment by julienschmidt** [comment]
If you made a binary dump, e.g. with [mongodump](https://docs.mongodb.org/manual/reference/program/mongodump/), it also contains the characters DB.
I think I have 2 or three DB dumps.

**Comment by marcusnovotny** [comment]
It would be amazing if we could get this done, that should be the most thorough fix. How complicated is that?

**Comment by sacdallago** [comment]
Side note:
we hit the limit
![screen shot 2016-04-04 at 10 14 11 pm](https://cloud.githubusercontent.com/assets/6240943/14261819/b7d21846-fab2-11e5-9eb3-64b1e723e8cc.png)

``` bash
[crawler/mobile] Belwas found: 100 inserted: 0 2016-02-12 00:11:38
[crawler/mobile][INFO] [1508/1510] MCRWLD Belwas { found: 100, inserted: 0 }
[crawler/mobile][INFO] [1509/1510] MCRWLD Wyl Waynwood { found: 0, inserted: 0 }
[crawler/mobile][INFO] [1510/1510] MCRWLD Norne Goodbrother { found: 0, inserted: 0 }
[main][INFO] Crawling completed
[main][INFO] Update complete!
[main][INFO] Scheduling next incremental update atMon Apr 04 2016 23:08:53 GMT+0200 (CEST)
```

Which I think is because it can't actually write anymore on the DB?

**Comment by julienschmidt** [comment]
File size is more than twice the data size? what?

**Comment by sacdallago** [comment]
:D Misteries

**Comment by marcusnovotny** [comment]
I think I found the problem @julienschmidt 
I followed your advice and changed the found & retryThreshold. Then I noticed the crawler always retries with the init date (1.1.1970)
![](http://i.imgur.com/leonKqY.png)

https://github.com/Rostlab/JS16_ProjectD_Group4/blob/develop/crawler/mobile.js#L162

lastDay is initially set to null, so when we decrease lastDay by one day in line 196 we run into problems. I changed the initial value to today's date and after skipping the first few tweets we already had the crawler is now crawling Jon Snow just like it should.

How can we initially set lastDay to the date of our oldest tweet? That should fix it
